src/multivariate/dirichlet.jl

```diff
@@ -257,15 +252,16 @@ function dirichlet_mle_init(P::AbstractMatrix{Float64}, w::AbstractArray{Float64
     K = size(P, 1)
     n = size(P, 2)
 
+    n == length(w) || throw(DimensionMismatch("Inconsistent argument dimensions."))
+
     μ = zeros(K)  # E[p]
     γ = zeros(K)  # E[p^2]
     tw = 0.0
 
-    for i in 1:n
-        @inbounds wi = w[i]
+    for (iP, wi) in zip(axes(P, 2), w)
         tw += wi
-        for k in 1:K
-            pk = P[k, i]
+        for k in axes(P, 1)
+            pk = P[k, iP]
             @inbounds μ[k] += pk * wi
             @inbounds γ[k] += pk * pk * wi
```

Comment:
Should use a different index for `μ` and `γ` as these may not have the same indices as `P`.